celery/app/task.py

```diff
@@ -846,20 +846,20 @@ def replace(self, sig):
 
```

Comment:
The branch above is never entered during the test suite, i.e. `'chord' in sig.options` is never true. I think this makes sense - for it to be true, we'd have to be replacing a task in a chord with another task that is a part of a different chord. I'm not sure how that would work - it might be worth just raising an exception in this branch rather than having this undefined untested behaviour in there.